docs/COMPONENTS.rst

```diff
@@ -21,47 +21,51 @@ a scheduler (a.k.a. master node), worker(s), package builder
 
 In the figure above, every service appears on a separate machine but this is
 not a requirement (see :ref:`lab_setup` for an alternative setup). Jobs are
-submitted to the scheduler, which are then processed by workers. One worker
-processes and keeps track of a job (one at a time). The output of the job (logs
-and files associated to an execution) is stored in the archive, which
-is a path in the file system where the worker is running. The job database
-contains information about the status of jobs and test nodes, as well as
-results of executions (# of tests passed, failed, etc.). All this information
-can be visualized in ``pulpito``, the web UI. For an example, see Ceph
-community's Lab `here <http://pulpito.ceph.com>`__.
+submitted to the scheduler, which are then picked up by dispatcher and
+processesed by job supervisors. One supervisor processes and keeps track of a
+job (one at a time). The output of the job (logs and files associated to an
+execution) is stored in the archive, which is a path in the file system where
+the dispatcher is running. The job database contains information about the status
+of jobs and test nodes, as well as results of executions (# of tests passed,
+failed, etc.). All this information can be visualized in ``pulpito``, the web
+UI. For an example, see Ceph community's Lab `here <http://pulpito.ceph.com>`__.
 
-Test nodes can be grouped in classes (referred to as ``machine-type``), 
-allowing teuthology schedule jobs across multiple hardware setups, 
+Test nodes can be grouped in classes (referred to as ``machine-type``),
+allowing teuthology schedule jobs across multiple hardware setups,
 provided they're visible to the master node.
 
 Life of a Teuthology Job
 ========================
 
 The teuthology scheduler exposes a work queue (using `beanstalkd
-<https://kr.github.io/beanstalkd/>`__) where jobs are submitted. The life of a 
-job begins when ``teuthology-suite`` is executed, which is when a job is 
-prepared and queued (``teuthology-schedule`` is implicitly invoked). When a job 
-is created (or whenever the status of a job is changed, e.g. from queued to 
-started), information about the job is recorded in ``paddles``'s internal 
-database. Depending on the requirements (in terms of requested machines), the 
-scheduler eventually determines when a job can get executed. At this point the 
-master communicates with ``teuthology-worker``, which in turn invokes 
```

Comment:
@jdurgin so with this change you guys are dropping any reference to teuthology-worker?

Reply:
yes, I think it'll be obsolete once the dispatcher si stabilized